project/src/main/java/com/google/autograder/servlets/GetAssignmentDetailsServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.autograder.servlets;
+
+import java.net.URL;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.autograder.servlets.helpers.Services;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+@WebServlet("/getAssignmentDetails")
+public final class GetAssignmentDetailsServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestURL = (String) request.getHeader("Referer");
+        String courseID = requestURL.substring(requestURL.indexOf("?courseID=") + 10);
```

Comment:
Does it work?

I imagine the url would be blah.google.com/getAssigmentDetails?courseID=abc&assignmentID=1234, right?

There won't be two '?' in the requestURL, am I right?

Reply:
Oh, yes you're right here, I'll have to go back and fix this.

Reply:
I think we can use request.getParameterValues("courseID") to retrieve the value instead of using sub string.
https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)